staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -505,14 +505,12 @@ func (f *DeltaFIFO) Replace(list []interface{}, resourceVersion string) error {
 
 // Resync will send a sync event for each item
 func (f *DeltaFIFO) Resync() error {
-	var keys []string
-	func() {
-		f.lock.RLock()
-		defer f.lock.RUnlock()
-		keys = f.knownObjects.ListKeys()
-	}()
+	f.lock.Lock()
+	defer f.lock.Unlock()
```

Comment:
@ncdc - IIUC, in case of PV not evertthing can be watch based (i.e. cloud providers usually don't expose watch on cloud resources). That said, to ensure that a PV, or LB or sth like that didn't change in the meantime (or wasn't deleted or sth), we need to periodically go over these. And IIUC, we are doing this by iterating over k8s resources and checking what is happening to the external resource associated with this.

That said, I'm afraid for those controllers that are controlling external resource, I'm not sure if we can reduce resync period to something really small (though there are probably a bunch of already existing in which we should be able to do that).